### PR TITLE
1 main code

### LINKED ISSUE #1: Create main_data_processor
A module that can be called from the worker.py to do the actual processing

### PR DESCRIPTION
**Pull request by greenw0lf**
Closes #1 

Added the main code that runs Whisper on audio to generate transcriptions. 

When testing, I recommend using the CPU for processing (it is already set to that in the config.yml). If you happen to have a more powerful Nvidia GPU available, then you can change the value to `cuda` and test it (I will also test it myself using a GPU I have available).

I have also already tested the `vad` and `word_timestamps` settings and it seems to be working fine.

### CODE REVIEW

File: whisper.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            "float16" if cfg.WHISPER_ASR_SETTINGS.DEVICE == "cuda" else "float32"
+        ),
+    )
+
+    segments, _ = model.transcribe(
+        input.input_file_path,
+        vad_filter=cfg.WHISPER_ASR_SETTINGS.VAD,
+        beam_size=5,
+        best_of=5,
+        temperature=(0.0, 0.2, 0.4, 0.6, 0.8, 1.0),
```
Comment (Veldhoen) [inline whisper.py:40]: Would it make sense to make these hard-coded values configurable? If not, a bit of explanation would be good to add.

File: whisper.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    )
+
+    segments_to_add = []
+    for segment in segments:
+        words_to_add = []
+        if cfg.WHISPER_ASR_SETTINGS.WORD_TIMESTAMPS:
+            for word in segment.words:
+                words_to_add.append(
+                    {
+                        "text": word.word.lstrip(),
```
Comment (Veldhoen) [inline whisper.py:52]: Why lstrip and not strip?

File: whisper.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    end = time.time() * 1000  # convert to ms
+
+    model_application_provenance = Provenance(
+        activity_name="whisper_asr",
+        activity_description="transcribed an audio file using Whisper",
+        input_data=input.input_file_path,
+        start_time_unix=start,
+        parameters=cfg.WHISPER_ASR_SETTINGS,
+        software_version="0.1.0",
```
Comment (Veldhoen) [inline whisper.py:87]: A hardcoded software version is not ideal

**Review by Veldhoen — COMMENTED** [review]
Looks good, and runs on my machine. Nice work! I added some small remarks, other than that ready to merge, I'd say :)

**Comment by Veldhoen** [comment]
Oh and of course it would be really great to add some automated tests, for instance for the config settings.

File: whisper.py (hunk 1 above)
Comment (greenw0lf) [reply whisper.py:40]: Yeah, making them configurable makes sense. As for where people can read more about this, I am not sure exactly. I think the official page for OpenAI's Whisper is a good starting point.

File: whisper.py (hunk 3 above)
Comment (greenw0lf) [reply whisper.py:87]: What should I set this to then?

File: whisper.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+) -> WhisperASROutput:
+    logger = logging.getLogger(__name__)
+    logger.info("Starting model application")
+    start = time.time() * 1000  # convert to ms
+    destination = get_output_file_path(input.source_id, OutputType.TRANSCRIPT)
+
+    # float16 only works on GPU, float32 or int8 are recommended for CPU
+    model = WhisperModel(
+        cfg.WHISPER_ASR_SETTINGS.MODEL_VERSION,
+        device=cfg.WHISPER_ASR_SETTINGS.DEVICE,
```
[the remark this answers is not in the document]
Comment (greenw0lf) [reply whisper.py:30]: Note to myself to look into it

**Pull request merged**